app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -217,4 +218,24 @@ def parse_vapp_template(vapp_template)
 
     return uid, new_result
   end
+
+  def parse_snapshot(vm)
+    snapshot_resp = @connection.get_snapshot_section(vm.id).data[:body][:Snapshot]
```

Comment:
Use `data.fetch_path(:body, :Snapshot)` here so we fail the refresh with an exception if body is somehow nil